presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/TarController.java

```diff
@@ -0,0 +1,235 @@
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to set launcher script path", e);
+        }
+    }
+
+    public void uninstallAsync(boolean checkDependencies)
+            throws PrestoManagerException
+    {
+        if (!checkDependencies) {
+            throw new PrestoManagerException("Unsupported parameter 'checkDependencies' for tarball uninstall");
+        }
+        try {
+            deleteRecursively(installationDir);
+            deleteRecursively(dataDir);
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to uninstall Presto", e);
+        }
+        LOGGER.debug("Successfully uninstalled Presto");
+    }
+
+    public void upgradeAsync(URL packageUrl, boolean checkDependencies, boolean preserveConfig)
+            throws PrestoManagerException
+    {
+        if (!checkDependencies) {
+            throw new PrestoManagerException("Unsupported parameter 'checkDependencies' for tarball upgrade");
+        }
+        Path tempPackage = getTarPackage(packageUrl);
+        try {
+            if (preserveConfig) {
+                Path tempConfig = configUtils.backupDirectory(configDir);
+                try {
+                    uninstallAsync(checkDependencies);
+                    tarInstall(tempPackage);
+                    configUtils.restoreDirectory(tempConfig, configDir);
+                }
+                finally {
+                    deleteTempFile(tempConfig);
+                }
+            }
+            else {
+                uninstallAsync(checkDependencies);
+                tarInstall(tempPackage);
+                configUtils.deployDefaultConfig(configDir, dataDir, pluginDir);
+                configUtils.deployDefaultConnectors(catalogDir);
+            }
+        }
+        finally {
+            deleteTempFile(tempPackage);
+        }
+        LOGGER.debug("Successfully upgraded presto");
+    }
+
+    public void startAsync()
+            throws PrestoManagerException
+    {
+        execLauncherCommand("start");
+    }
+
+    public void terminate()
+            throws PrestoManagerException
+    {
+        execLauncherCommand("stop");
+    }
+
+    public void kill()
+            throws PrestoManagerException
+    {
+        execLauncherCommand("kill");
+    }
+
+    public void restartAsync()
+            throws PrestoManagerException
+    {
+        execLauncherCommand("restart");
+    }
+
+    private void execLauncherCommand(String command)
+            throws PrestoManagerException
+    {
+        int commandResult = executor.runCommand(launcherScript.toAbsolutePath().toString(), command);
+        if (commandResult != 0) {
+            throw new PrestoManagerException(format("Failed to %s Presto", command), commandResult);
+        }
+    }
+
+    public String getVersion()
+            throws PrestoManagerException
+    {
+        throw new PrestoManagerException("Unsupported operation");
```

Comment:
If Presto is running I get version by querying Presto. This method is called only when Presto is installed and not running.